File: DesktopSample/DataBinding/Section/XML/VB.NET/XML/StartForm.vb
Imports System
Imports System.Text
Imports System.Windows.Forms
Imports System.Xml
Imports GrapeCity.ActiveReports
Imports GrapeCity.ActiveReports.SectionReportModel

'XMLデータソースを使い、受注一覧レポートを作成します。
' サブレポートを使う方法と、XML階層構造を使用してレポートを作成する方法を示します。
Public Class StartForm
	Inherits System.Windows.Forms.Form

	Public Sub New()
		MyBase.New()
		Encoding.RegisterProvider(CodePagesEncodingProvider.Instance)
		' 必要なデザイナ変数です。
		InitializeComponent()
	End Sub

	'Clean up any resources being used.
	Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
		If disposing Then
			If Not (components Is Nothing) Then
				components.Dispose()
			End If
		End If
		MyBase.Dispose(disposing)
	End Sub

	'btnCustomers_Clickイベントでラジオボタンを選択し、レポートのオブジェクトのデータを設定します。
	Private Sub btnCustomers_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnCustomers.Click
		Try
			'Dim rpt As New CustomersOrders()
			Dim rpt As New SectionReport()
			rpt.LoadLayout(XmlReader.Create(My.Resources.CustomersOrders))

			CType(rpt.DataSource, Data.XMLDataSource).FileURL = My.Resources.ConnectionString
			If radioAll.Checked Then 'データをすべて表示します。
				CType(rpt.DataSource, Data.XMLDataSource).RecordsetPattern = "//CUSTOMER"
			ElseIf radioID.Checked Then 'ID=2301に設定されたデータを表示します。
				CType(rpt.DataSource, Data.XMLDataSource).RecordsetPattern = "//CUSTOMER[@id = " + """" + "2301" + """" + "]"
			ElseIf radioEmail.Checked Then '有効なE-mailを含むデータを表示します。
				CType(rpt.DataSource, Data.XMLDataSource).RecordsetPattern = "//CUSTOMER[@email]"
			End If

			Dim frm As New ViewerForm()
			frm.Show()
			frm.LoadReport(rpt)
		Catch ex As ReportException
			MessageBox.Show(ex.ToString(), Text)
		End Try
	End Sub 'btnCustomers_Click

	'btnCustomersLeveled_Clickイベントで OrdersLeveledレポートを作成し、
	' データソースを設定します。
	Private Sub btnCustomersLeveled_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnCustomersLeveled.Click
		Dim rpt As New SectionReport()
		rpt.LoadLayout(XmlReader.Create(My.Resources.OrdersLeveled))
		CType(rpt.DataSource, Data.XMLDataSource).FileURL = My.Resources.ConnectionString
		rpt.Run()
		Dim frm As New ViewerForm()
		frm.Show()
		frm.LoadReport(rpt)
	End Sub 'btnCustomersLeveled_Click

End Class 'StartForm

File: DesktopSample/DataBinding/Section/XML/C#/XML/StartForm.cs
using GrapeCity.ActiveReports;
using System;
using System.Text;
using System.Windows.Forms;
using System.Xml;

namespace ActiveReports.Samples.XML
{
	/// <summary>
	/// XMLデータソースを使い、受注一覧レポートを作成します。
	/// サブレポートを使う方法と、XML階層構造を使用してレポートを作成する方法を示します。
	/// </summary>
	public partial class StartForm : System.Windows.Forms.Form
	{
		
		public StartForm()
		{
			// Windows フォーム デザイナ サポートに必要です。
			//
			InitializeComponent();
		}
	  
		/// <summary>
		/// アプリケーションのメインエントリポイントです。
		/// </summary>
		[STAThread]
		static void Main() 
		{
            Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
            Application.EnableVisualStyles();
            Application.SetCompatibleTextRenderingDefault(false);
#if NET6_0_OR_GREATER
			Application.SetHighDpiMode(HighDpiMode.DpiUnawareGdiScaled);
#endif
            Application.Run(new StartForm());
		}

		/// <summary>
		/// btnCustomers_Clickイベントでラジオボタンを選択し、レポートのオブジェクトのデータを設定します。
		/// </summary>
		private void btnCustomers_Click(object sender, EventArgs e)
		{
			try
			{
				var rpt = new SectionReport();
				rpt.LoadLayout(XmlReader.Create(Properties.Resources.CustomersOrders));

				var ds = rpt.DataSource as GrapeCity.ActiveReports.Data.XMLDataSource;
				if (ds == null)
				{
					// エラーが発生した場合、メッセージを表示します。
					MessageBox.Show(Properties.Resources.DataSourceError, this.Text);
					return;
				}

				ds.FileURL = Properties.Resources.ConnectionString;

				// 作成するレポートに合わせて、ノード（レコード）を取得するためのXSLパターンを設定します。
				if (radioAll.Checked)
				{
					// データをすべて表示します。
					ds.RecordsetPattern = "//CUSTOMER";
				}
				else if (radioID.Checked)
				{
					// ID=2301に設定されたデータを表示します。
					ds.RecordsetPattern = "//CUSTOMER[@id = " + @"""" + "2301" + @"""" + "]";
				}
				else if (radioEmail.Checked)
				{
					// 有効なE-mailを含むデータを表示します。
					ds.RecordsetPattern = "//CUSTOMER[@email]";
				}
				
				ViewerForm frm = new ViewerForm();
				frm.Show();
				frm.LoadReport(rpt);
			}
			catch (ReportException ex)
			{
				MessageBox.Show(ex.ToString(), Text);
			}
		}

		/// <summary>
		/// btnCustomersLeveled_Clickイベントで OrdersLeveledレポートを作成し、
		/// データソースを設定します。
		/// </summary>
		private void btnCustomersLeveled_Click(object sender, EventArgs e)
		{
			try
			{
				//OrdersLeveled rpt = new OrdersLeveled();
				var rpt = new SectionReport();
				rpt.LoadLayout(XmlReader.Create(Properties.Resources.OrdersLeveled));

				var ds = rpt.DataSource as GrapeCity.ActiveReports.Data.XMLDataSource;
				if (ds == null)
				{
					// エラーが発生した場合、メッセージを表示します。
					MessageBox.Show(Properties.Resources.DataSourceError, this.Text);
					return;
				}

				// XMLファイル名を設定します。
				ds.FileURL = Properties.Resources.ConnectionString;
				
				// レポートを表示します。
				ViewerForm frm = new ViewerForm();
				frm.Show();
				frm.LoadReport(rpt);
			}
			catch (ReportException ex)
			{
				MessageBox.Show(ex.ToString(), Text);
			}
		}
	}
}
